sbin/wifibox

```diff
@@ -734,8 +740,22 @@ vm_manager() {
 
     log debug "Nice priority: ${_nice_priority}"
     log debug "Arguments: ${_bhyve_args}"
-    ${NICE} -n ${_nice_priority} \
-	    ${BHYVE} ${_bhyve_args} 2>&1 | capture_output debug bhyve
+
+    _kill_child() {
+	_jobs="$(jobs -p)"
+
+	[ -n "$_jobs" ] && ${KILL} -SIGTERM $_jobs
+	exit 0
+    }
+
+    trap _kill_child SIGTERM
+
+    { ${NICE} -n ${_nice_priority} \
+	    ${BHYVE} ${_bhyve_args} 2>&1 | capture_output debug bhyve; } &
+    wait
+
+    trap '' SIGTERM
+
```

Comment:
We will need to second `wait` with the identifier of the forked process to get its exit code otherwise the `_bhyve_exit_code` below is not going to be populated properly, which is used in the rest of the function.

Reply:
How about this instead?

```    trap _kill_child SIGTERM

    { ${NICE} -n ${_nice_priority} \   
            ${BHYVE} ${_bhyve_args} 2>&1 | capture_output debug bhyve; } &
    wait $!
    _bhyve_exit_code="$?"
    
    trap - SIGTERM
        
    destroy_vm
```

Reply:
Based on my testing so far this does not work either, unfortunately.  We will need the PID-specific second `wait` after `trap` otherwise the exit code does not propagate.  This is what the manager uses to decide to restart the VM or let it stop.

Reply:
I'm not sure I follow.

> We will need the PID-specific second wait after trap otherwise the exit code does not propagate

Propagate to/from where? 

The following mockup:

```
#!/bin/
fun() {
  echo hello
}

trap fun SIGTERM

{ sleep 5 | fun; } &
wait $!
huh="$?"

trap - SIGTERM

echo "$huh"
```

when run use `sh test.sh`, and then killed using `kill -SIGTERM $pid` where $pid = the PID of `sh` prints "hello 143" for me, as expected. If `sleep` is killed, it prints "hello 0". AFAICT, the latter case is not relevant here, because `bhyvectl` is not being killed by `daemon`, `/bin/sh /usr/local/sbin/wifibox _manage_vm` is.

Reply:
Oh, right, we need to wait for a second time because we need to wait for the process to end after being killed.

Reply:
Hopefully it's all fine now.